lib/pf/mfa/TOTP.pm

```diff
@@ -118,7 +121,9 @@ sub redirect_info {
     my $logger = get_logger();
     $logger->info("MFA USERNAME: ".$username);
     my ($exist, $otp) = $self->generate_otp($username);
-
+    # Set in the cache that the user did the redirection on Akamai MFA
+    cache->set($username."mfapreauth", $TRUE, normalize_time($self->cache_duration) * 2);
+    cache->set($username."mfapostauth", $FALSE, normalize_time($self->cache_duration) * 2);
```

Comment:
I'm going to be mr. picky but this is copy/pasted between the 2 MFA modules, please centralize the logic of success/failure MFA calls in the cache to the base MFA module